### PR TITLE
Add test for single-file net6

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs [hunk 1]
```diff
@@ -253,6 +253,30 @@ public void It_generates_a_single_file_for_self_contained_apps()
                 .NotHaveFiles(unexpectedFiles);
         }
 
+        [RequiresMSBuildVersionFact("16.8.0")]
+        public void No_runtime_files_6_0()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "net6.0",
```
Comment (agocke) [inline src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:262]: @wli3 is there some way to grab the id of the newest target framework so I can use it here? I want this test to catch any regressions, including ones we make in 7.0, etc

Comment (agocke) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:262]: also @dsplaisted

Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:262]: If you want the test to use the latest version of the framework, I think the closest way to do that would be to shell out to `dotnet new console`.  Then whenever the templates were updated, the test would also update.

Note that that's also not a guarantee- we could decide that the templates should target an LTS version.

Comment (agocke) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:262]: That seems a bit heavy-handed, I was hoping there could just be a global string that was either generated or updated every year on the new release.

It's a bit concerning that this isn't a common pattern. Does this mean that the vast majority of the tests in the repo are written for the current/previous TFs and then never updated? So we have very little coverage of the new TFs?

**Comment by VSadov** [comment]
I think the fix is in SDK now. Perhaps worth rebasing and trying again?

**Pull request merged**